scrapper.py

```diff
@@ -26,38 +41,183 @@ class Crawler:
+
+    def _fill_article_with_meta_information(self, article_bs):
+        material_before_body = article_bs.find('div', class_='b-material-before-body')
+
+        title_bs = material_before_body.find('h1')
+        self.article.title = title_bs.text
+
+        topic_bs = material_before_body.find('div', class_='b-material__rubrics').find('a')
+        self.article.topics = [topic_bs.text]
+
+        tags_bs = material_before_body.find('div', class_='b-material__tags')
+        author = 'NOT FOUND'
+
+        for tag_bs in tags_bs:
+            match = re.match(r'[А-Я][а-я]+\s[А-Я][а-я]+', tag_bs.text)
+
+            if match:
+                author = tag_bs.text
+                break
+
+        self.article.author = author
+
+        head_title_bs = article_bs.find('title')
+        date_match = re.search(r'\d{2}\.\d{2}\.\d{4}', head_title_bs.text)
+        date_time_string = date_match.group(0)
+
+        time_bs = material_before_body.find('p', class_='b-material__date')
+        time_match = re.search(r'\d{2}:\d{2}', time_bs.text)
+        date_time_string = ' '.join([date_time_string, time_match.group(0)])
+
+        date_time = datetime.strptime(date_time_string, '%d.%m.%Y %H:%M')
+        self.article.date = date_time
 
 
 def prepare_environment(base_path):
     """
     Creates ASSETS_PATH folder if not created and removes existing folder
     """
-    pass
+    path = Path(base_path)
+    if path.is_dir():
+        shutil.rmtree(path)
+
+    path.mkdir(parents=True, exist_ok=True)
 
 
 def validate_config(crawler_path):
     """
     Validates given config
     """
-    pass
+    with open(crawler_path, 'r', encoding='utf-8') as file:
+        config = json.load(file)
+        seed_urls = config['seed_urls']
+        max_articles = config['total_articles_to_find_and_parse']
+
+        if not isinstance(seed_urls, list):
+            raise IncorrectURLError
+        # если переменная seed_urls не является списком
+
+        if not seed_urls:
+            raise IncorrectURLError
+        # возбуждает исключение, если такого seed_url
+        # нет в списке, который мы сделали в scrapper_config.json
+
+        if not isinstance(max_articles, int):
+            raise IncorrectNumberOfArticlesError
+        # если число статей, которое мы
+        # передаем в scrapper_config.json не является integer
+
+        if max_articles > 100000:
+            raise NumberOfArticlesOutOfRangeError
+        # если статей для парсинга больше 100000
+
+        if max_articles == 0 or max_articles < 0:
+            raise IncorrectNumberOfArticlesError
+        # если число статей, которое мы
+        # передаем в scrapper_config.json не является integer
+
+        for seed_url in seed_urls:
+            if seed_url[0:8] != 'https://' and seed_url[0:7] != 'http://':
+                raise IncorrectURLError
+        # если протокол не соответствует стандратному паттерну
+
+        prepare_environment(ASSETS_PATH)  # пока не починили тесты
```

Comment:
environment preparation is not a responsibility of config validator

Reply:
i fixed that